app/models/manageiq/providers/workflows/automation_manager/configuration_script_source.rb

```diff
@@ -18,9 +18,24 @@ def sync
           next if filename.start_with?(".") || !filename.end_with?(".asl")
 
           payload = worktree.read_file(filename)
-          found   = current.delete(filename) || self.class.module_parent::Workflow.new(:configuration_script_source_id => id)
-
-          found.update!(:name => filename, :manager_id => manager_id, :payload => payload, :payload_type => "json")
+          payload_valid, payload_error =
+            begin
+              Floe::Workflow.new(payload)
```

Comment:
If the payload is _very_ broken, like bad JSON broken, will it still raise a Floe::InvalidWorkflowError ?

Reply:
Yes https://github.com/ManageIQ/floe/blob/master/lib/floe/workflow.rb#L57-L58